srv/salt/ceph/apparmor/install.sls

```diff
@@ -0,0 +1,6 @@
+
+zypper -n in -t pattern apparmor:
```

Comment:
I don't suppose `pkg.installed` is viable here to avoid the `zypper` SUSE-ism? I'm pretty sure you can use something like `patterns-base-apparmor` as the package name with `pattern.installed` (but haven't tested).

Reply:
ahh thx...will try it out.

Reply:
With zypper it's poosible to do ```zypper in +pattern:apparmor```. You'll be shocked to learn that pkg.install does not understand this. Otherwise pkg is aware of patterns (can list them) but I couldn't find a way to install one.

Reply:
I thought it might work if you did `pkg.installed patterns-base-apparmor`, because a pattern these days is somehow kinda just a magic RPM that Requires a bunch of other RPMs....

Reply:
Weird. I just tried here on a SLE 15 VM:

````
# salt admin.ceph pkg.install patterns-base-apparmor
admin.ceph:
    ----------
    apparmor-abstractions:
        ----------
        new:
            2.12-3.13
        old:
    apparmor-docs:
        ----------
        new:
            2.12-3.13
        old:
    apparmor-parser:
        ----------
        new:
            2.12-3.13
        old:
    apparmor-parser-lang:
        ----------
        new:
            2.12-3.13
        old:
    apparmor-profiles:
        ----------
        new:
            2.12-3.13
        old:
    apparmor-utils:
        ----------
        new:
            2.12-3.13
        old:
    apparmor-utils-lang:
        ----------
        new:
            2.12-3.13
        old:
    augeas:
        ----------
        new:
            1.9.0-1.7
        old:
    gpg-pubkey:
        ----------
        new:
            50a3dd1c-50f35137
        old:
            307e3d54-4be01a65,39db7c82-5847eb1f,50a3dd1c-50f35137
    kernel-default:
        ----------
        new:
            4.12.14-12.2
        old:
            4.12.14-16.4.1,4.12.14-12.2
    libjemalloc2:
        ----------
        new:
            5.0.1-1.14
        old:
    libyui-ncurses8:
        ----------
        new:
            2.50.1-1.4
        old:
    libyui8:
        ----------
        new:
            3.4.0-1.15
        old:
    patterns-base-apparmor:
        ----------
        new:
            20171206-9.3
        old:
    perl-Config-Crontab:
        ----------
        new:
            1.45-1.14
        old:
    perl-XML-NamespaceSupport:
        ----------
        new:
            1.12-1.13
        old:
    perl-XML-SAX:
        ----------
        new:
            0.99-1.11
        old:
    perl-XML-SAX-Base:
        ----------
        new:
            1.09-1.14
        old:
    perl-XML-SAX-Expat:
        ----------
        new:
            0.51-1.11
        old:
    perl-XML-Simple:
        ----------
        new:
            2.24-1.11
        old:
    perl-apparmor:
        ----------
        new:
            2.12-3.13
        old:
    python3-apparmor:
        ----------
        new:
            2.12-3.13
        old:
    ruby:
        ----------
        new:
            2.5-1.9
        old:
    ruby2.5-rubygem-abstract_method:
        ----------
        new:
            1.2.1-1.16
        old:
    ruby2.5-rubygem-cfa:
        ----------
        new:
            0.6.3-1.3
        old:
    ruby2.5-rubygem-cheetah:
        ----------
        new:
            0.5.0-1.16
        old:
    ruby2.5-rubygem-fast_gettext:
        ----------
        new:
            1.6.0-1.6
        old:
    ruby2.5-rubygem-ruby-augeas:
        ----------
        new:
            0.5.0-1.19
        old:
    ruby2.5-rubygem-ruby-dbus:
        ----------
        new:
            0.14.0-1.11
        old:
    ruby2.5-rubygem-simpleidn:
        ----------
        new:
            0.0.9-1.5
        old:
    yast2:
        ----------
        new:
            4.0.59-1.2
        old:
    yast2-apparmor:
        ----------
        new:
            4.0.4-1.2
        old:
    yast2-core:
        ----------
        new:
            4.0.2-1.5
        old:
    yast2-hardware-detection:
        ----------
        new:
            3.1.8-1.17
        old:
    yast2-perl-bindings:
        ----------
        new:
            4.0.1-1.4
        old:
    yast2-pkg-bindings:
        ----------
        new:
            4.0.9-1.5
        old:
    yast2-ruby-bindings:
        ----------
        new:
            4.0.4-1.7
        old:
    yast2-snapper:
        ----------
        new:
            4.0.1-1.8
        old:
    yast2-xml:
        ----------
        new:
            3.3.0-1.15
        old:
    yast2-ycp-ui-bindings:
        ----------
        new:
            4.0.0-1.13
        old:
````

Reply:
Huh weird indeed. I did try this on a salt 2016 cluster...will double check for salt 17/18 and include this. thx.

Reply:
ok, seems like this is not a salt issue but SLE15 now has a dummy package (patterns-base-apparmor) that SP3 did not have. Tumbleweed has it too, so I assume so does Leap 15. I.e. I'll change this to the more generic pkg.installed call. Does anyone know if Centos has a similar meta package? @rjfd maybe? For Ubuntu it seems ```apparmor``` and ```apparmor-utils``` do the trick.